lib/ansible/module_utils/facts/system/pkg_mgr.py

```diff
@@ -82,7 +82,13 @@ def _check_rh_versions(self, pkg_mgr_name, collected_facts):
                 # just default to dnf
                 pkg_mgr_name = 'dnf'
         elif collected_facts['ansible_distribution'] == 'Amazon':
-            pkg_mgr_name = 'yum'
+            try:
+                if int(collected_facts['ansible_distribution_major_version']) < 2022:
```

Comment:
Better, but that string was just split on `.`, so there's still no guarantee it's parseable as an int, and the consequences of a potential future failure will break all fact collection for that distro, so extra-defensive is better. eg: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/facts/system/pkg_mgr.py#L70-L83

(and yes, I know there are other places that aren't doing it as carefully, but we shouldn't be making more of them ;) )

Reply:
Hmm not sure I completely follow, but I adapted the code to be more analogue to the above Fedora implementation:

* if its not a parsable major version, assume that it will be `dnf` going forward
* if its a version below 2022 (including 2018 and AL2), then check if yum exists and only then select it
* any higher version, check if dnf exists, then select it

Let me know though if I am misunderstanding what this should be doing/returning :)